src/main/java/uk/gov/cshr/service/security/IdentityService.java

```diff
@@ -98,6 +98,14 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
         return new IdentityDetails(identity);
     }
 
+    public String getUIDFromEmail(String email) {
+        Optional<Identity> identity = identityRepository.findIdentityByEmailEquals(email);
+        if(identity.isPresent()) {
+            return identity.get().getUid();
+        }
+        return null;
+    }
+
```

Comment:
Above code can be simplified by using Java 8 style coding.

Reply:
can discuss this further.